x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -626,6 +628,15 @@ private static boolean checkGroupMatch(Expression e, Node<?> source, List<Expres
         return false;
     }
 
+    private static void checkBooleanFiltering(LogicalPlan p, Set<Failure> localFailures) {
+        if (p instanceof Filter) {
+            Expression condition = ((Filter) p).condition();
+            if (condition.resolved() && condition.dataType() != BOOLEAN) {
+                localFailures.add(fail(condition, "Cannot filter by non-boolean expression of type [{}]", condition.dataType()));
```

Comment:
Maybe `Condition expression needs to be boolean, found [{}]`. Remove the notion of filter which is not something the user declares.

Reply:
Changed.